### ISSUE TITLE
Build failed: expected tuple, found `()`

### ISSUE THREAD

**Issue by zoechi** [rust-db/barrel#96]
After I run into https://github.com/rust-db/barrel/issues/94 I tried to install from Github
but I'm currently stuck at

```sh
cargo install --git https://github.com/diesel-rs/diesel --features barrel-migrations diesel_cli
```

```
```
   Compiling barrel v0.6.5
error[E0308]: mismatched types
   --> /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/barrel-0.6.5/src/integrations/diesel.rs:190:26
    |
190 | fn get_backend_pair() -> (&'static str, &'static str) {
    |    ----------------      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected tuple, found `()`
    |    |
    |    implicitly returns `()` as its body has no tail or `return` expression
    |
    = note:  expected tuple `(&'static str, &'static str)`
            found unit type `()`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: could not compile `barrel`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `diesel_cli v1.4.0 (https://github.com/diesel-rs/diesel#d9e4ffcf)`, intermediate artifacts can be found at `/tmp/cargo-installJJXITF`

Caused by:
  build failed
```
active toolchain
----------------

nightly-x86_64-unknown-linux-gnu (default)
rustc 1.43.0-nightly (c20d7eecb 2020-03-11)
```

**Comment by omid** [comment]
To bypass this error you can temporarily run the following command:

`cargo install --git https://github.com/diesel-rs/diesel --features "barrel-migrations,barrel/pg" diesel_cli`

By the way, you may face some other problems, since you are using the master branch of diesel.

**Comment by zoechi** [comment]
Thanks, I got it working with refinery after applying the open PR, that is fine for now.

**Comment by pksunkara** [comment]
Diesel definitely supports multiple databases at the same time. I think barrel should do too. The only blocker I can see for this is, how does the [`SqlGenerator`](https://docs.rs/barrel/0.6.5/barrel/backend/trait.SqlGenerator.html) know which database to generate for. We could add a config field in diesel.toml.

Or can we get it from diesel? The trait's [method](https://docs.rs/diesel_migrations/1.4.0/diesel_migrations/trait.Migration.html#tymethod.run) we need to implement has `&dyn SimpleConnection` as argument. Can we  somehow get the backend from that?

**Comment by zoechi** [comment]
@pksunkara looks like your comment is not related to this issue.

**Comment by pksunkara** [comment]
Nope, I meant to comment on this issue. What I am saying is that we should not need to define feature `barrel/pg` while installing `diesel`. That is relevant to this issue because without that feature, people would get this error. My comment is about how to fix this issue with future in mind.

**Comment by pksunkara** [comment]
I have fixed this in [this](https://github.com/reign-rs/barrel/commit/2043dd81cd66a0743318c94d72c5c5b6ccc358be) commit and the related PR to diesel is [here](https://github.com/diesel-rs/diesel/pull/2409)
